packages/api-v2/src/ws-client/subscriptions-manager.ts

```diff
@@ -0,0 +1,60 @@
+import Subscription from './subscription';
+import { TSocketResponse, TSocketRequestPayload, TSocketSubscribableEndpointNames } from '../../types';
+import { getQueryKeys } from '../utils';
+
+export default class SubscriptionsManager {
+    backendSubscriptions: Map<string, Subscription> = new Map();
+    authorizedWs: WebSocket | undefined;
+
+    setAuthorizedWs(authorizedWs: WebSocket) {
+        this.authorizedWs = authorizedWs;
+
+        this.backendSubscriptions.forEach(subscription => {
+            subscription.setAuthorizedWs(authorizedWs);
+        });
+    }
+
+    async subscribe(
+        name: TSocketSubscribableEndpointNames,
+        payload: TSocketRequestPayload<TSocketSubscribableEndpointNames>['payload'],
+        onData: (data: TSocketResponse<TSocketSubscribableEndpointNames>) => void
+    ) {
+        if (!this.authorizedWs) {
+            throw new Error('No connection');
+        }
+        const keys: Array<string> = getQueryKeys(name, payload); // potentially switch to separate function, which just returns primitive values (string)
+        const key = keys.join('-'); // but until then, use "join" to conver array of specifci structre to just string (no idea why its overcomplicated on first place)
+
+        let backendSubscription: Subscription | undefined;
+
+        if (!this.backendSubscriptions.has(key)) {
```

Comment:
Please correct me if I'm wrong, but from my understanding, we are storing all subscriptions inside backendSubscriptions (based on a key which is the endpoints name). Whenever there is already a subscription to that endpoint, we prevent it from happening again. However, for endpoints like `exchange_rates`, where different subscriptions with different payloads are allowed, are we handling that as well? I couldn't find the logic for that. Could you elaborate on it? Thank you.